xbmc/platform/freebsd/network/NetworkFreebsd.cpp

```diff
@@ -92,23 +91,18 @@ bool CNetworkInterfaceFreebsd::GetHostMacAddress(unsigned long host_ip, std::str
   struct rt_msghdr* rtm;
   struct sockaddr_inarp* sin;
   struct sockaddr_dl* sdl;
-  int mib[6];
+  constexpr std::array<int, 6> mib = {
+      CTL_NET, PF_ROUTE, 0, AF_INET, NET_RT_FLAGS, RTF_LLINFO,
+  };
 
   mac = "";
 
-  mib[0] = CTL_NET;
-  mib[1] = PF_ROUTE;
-  mib[2] = 0;
-  mib[3] = AF_INET;
-  mib[4] = NET_RT_FLAGS;
-  mib[5] = RTF_LLINFO;
-
-  if (sysctl(mib, ARRAY_SIZE(mib), NULL, &needed, NULL, 0) == 0)
+  if (sysctl(mib.data(), mib.size(), NULL, &needed, NULL, 0) == 0)
   {
     buf = (char*)malloc(needed);
     if (buf)
     {
-      if (sysctl(mib, ARRAY_SIZE(mib), buf, &needed, NULL, 0) == 0)
+      if (sysctl(mib.data(), mib.size(), buf, &needed, NULL, 0) == 0)
```

Comment:
Here the same and above, as already a change inside can also come `nullptr`

Reply:
done